ukis_csmask/mask.py

```diff
@@ -24,32 +24,30 @@ class CSmask:
     def __init__(
         self,
         img,
-        band_order=None,
+        band_order,
         nodata_value=None,
         invalid_buffer=4,
     ):
         """
         :param img: Input satellite image of shape (rows, cols, bands). (ndarray).
-            Requires images of Sentinel-2, Landsat-8, -7 or -5 in Top of Atmosphere reflectance [0, 1].
-            Requires image bands to include at least "Blue", "Green", "Red", "NIR", "SWIR1", "SWIR2".
-            Requires image bands to be in approximately 30 m resolution.
-        :param band_order: Image band order. (dict).
-            >>> band_order = {0: "Blue", 1: "Green", 2: "Red", 3: "NIR", 4: "SWIR1", 5: "SWIR2"}
+            Requires satellite images in Top of Atmosphere reflectance [0, 1].
+            Requires image bands to include at least "Blue", "Green", "Red", "NIR" (uses 4 band model).
+            For better performance requires image bands to include "Blue", "Green", "Red", "NIR", "SWIR1", "SWIR2" (runs 6 band model).
+            For better performance requires image bands to be in approximately 30 m resolution.
+        :param band_order: Image band order. (list of string).
+            >>> band_order = ["Blue", "Green", "Red", "NIR", "SWIR1", "SWIR2"]
         :param nodata_value: Additional nodata value that will be added to valid mask. (num).
-        :param invalid_buffer: Number of pixel that should be buffered around invalid areas.
+        :param invalid_buffer: Number of pixels that should be buffered around invalid areas.
         """
         # consistency checks on input image
-        if band_order is None:
-            band_order = ["Blue", "Green", "Red", "NIR", "SWIR1", "SWIR2"]
-
         if isinstance(img, np.ndarray) is False:
             raise TypeError("img must be of type np.ndarray")
 
         if img.ndim != 3:
             raise TypeError("img must be of shape (rows, cols, bands)")
 
-        if img.shape[2] < 6:
-            raise TypeError("img must contain at least 6 spectral bands")
+        if img.shape[2] < 4:
```

Comment:
Should we test more things here, like 5 or >6?

Reply:
It can be any number of bands large than 4. No need to specifically test for other things here. We also test the band_order to make sure the required spectral bands  are included.